src/components/transport_manager/CMakeLists.txt

```diff
@@ -43,60 +43,85 @@ include_directories (
+
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/transport_adapter/threaded_socket_connection.cc
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/transport_adapter/transport_adapter_impl.cc
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/transport_adapter/transport_adapter_listener_impl.cc
+
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/iap2_emulation/iap2_transport_adapter.cc
+
+)
+
+if (CMAKE_SYSTEM_NAME STREQUAL "QNX")
+  list(APPEND SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/platform_specific/qnx/platform_specific_network_interface_listener.cc
+  )
+elseif (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  list(APPEND SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/platform_specific/linux/platform_specific_network_interface_listener.cc
+  )
+endif ()
+
 if(BUILD_BT_SUPPORT)
   list(APPEND LIBRARIES
     bluetooth
   )
-else()
-  list(APPEND EXCLUDE_PATHS
-    ${COMPONENTS_DIR}/transport_manager/include/transport_manager/bluetooth
-    ${COMPONENTS_DIR}/transport_manager/src/bluetooth
+  list(APPEND SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth/bluetooth_connection_factory.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth/bluetooth_device.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth/bluetooth_device_scanner.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth/bluetooth_socket_connection.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth/bluetooth_transport_adapter.cc
   )
 endif()
 
 if(BUILD_USB_SUPPORT)
+  list(APPEND SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/usb_aoa_adapter.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/usb_connection_factory.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/usb_device_scanner.cc
+  )
   if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    set(EXCLUDE_PATHS
-      ${COMPONENTS_DIR}/transport_manager/include/transport_manager/usb/qnx
-      ${COMPONENTS_DIR}/transport_manager/src/usb/qnx
+    list(APPEND SOURCES
+      ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/libusb/platform_usb_device.cc
+      ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/libusb/usb_connection.cc
+      ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/libusb/usb_handler.cc
     )
-
   elseif(CMAKE_SYSTEM_NAME STREQUAL "QNX")
-    set(EXCLUDE_PATHS
-      ${COMPONENTS_DIR}/transport_manager/include/transport_manager/usb/libusb
-      ${COMPONENTS_DIR}/transport_manager/src/usb/libusb
-    )
     list(APPEND LIBRARIES
       usbdi
     )
+    list(APPEND SOURCES
+      ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/qnx/platform_usb_device.cc
+      ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/qnx/usb_connection.cc
+      ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/qnx/usb_handler.cc
+    )
   endif()
-else()
-  list(APPEND EXCLUDE_PATHS
-    ${COMPONENTS_DIR}/transport_manager/include/transport_manager/usb
-    ${COMPONENTS_DIR}/transport_manager/src/usb
-  )
-endif()
-
-if(NOT BUILD_TESTS)
-  list (APPEND EXCLUDE_PATH
-    ${CMAKE_CURRENT_SOURCE_DIR}/include/iap2_emulation/iap2_transport_adapter.h
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/iap2_emulation/iap2_transport_adapter.cc
-  )
 endif()
 
-collect_sources(SOURCES "${PATHS}" "${EXCLUDE_PATHS}")
```

Comment:
This is a change from Luxsoft team for adding QNX implementation. I guess they didn't like a logic such as:
```
if (build_is_for_linux) then
  exclude qnx related files
else if (build_for_qnx) then
  exclude linux related files
endif
```
... as this will be more complicated if we should add another target in future.

Let me see if I can keep original `collect_sources()` code while keeping the entire logic clean.

Reply:
I updated CMakeLists.txt file based on the original collect_sources() implementation. Please have a look at commit https://github.com/smartdevicelink/sdl_core/pull/2164/commits/a64d9247b09d633d089e01ace6749bec3c4db1b6.
Hi @LuxoftAKutsan, @askirk, could you also review if the updated file still works for your QNX build? Thanks in advance.

Reply:
@shoamano83 collect sources uses GLOB file command that is bad practice in CMake. 
The good practice is to list explicitly all fies that used for compilation. 
There are 2 resons: 
1. CMake does not know if file was changes and may not run cmake again during call make if you add additional source file to your project.
2. Not obvious for other developer what files actually used for compilation. Sometimes it leads to rather complicated linkage or compilation errors. function collect sources is recursive and it may take some sources that we didn't expect to get. 
On SYNC4 platform all CMake files was completely rewritten for using best practice of CMake 3.0 +

So this will not affect SYNC4 QNX compilation.